### PR TITLE
[nouislider]: Fix TSC error. Fix Callback signature (unencodedValues)

### PR DESCRIPTION
**Pull request by christophercr**
Fix TSC error: "global module exports may only appear at top level" (remove wrong duplicate namespace export).

Fix Callback interface: unencodedValues is an array of values instead of single value.

### CODE REVIEW

**Comment by DanielRosenwasser** [comment]
I don't understand - you're still going to get an error when using this as a global, so now nobody can use this module as a UMD global.

**Comment by christophercr** [comment]
Oh yeah, right, I overlooked that, I'll fix it to make it global but without having the TSC error...

**Comment by christophercr** [comment]
I've converted it to module global, so you can use it as a global (also UMD):

`import * as noUiSlider from "nouislider";
import noUiSlider = require("nouislider");`

**Comment by minestarks** [comment]
What TSC error are you trying to fix? Can you provide an example or a test? The imports work just as fine before and after your change, while using it as a global is now broken.

**Comment by christophercr** [comment]
Well, I solved a TSC error: "global module exports may only appear at top level" caused by the "export as namespace nouislider" line. I use typescript 2.3.x.

And thanks to comment from @DanielRosenwasser I made it a global module (you got an error when you tried it as global dependency, so before they were in fact a module dependency).

**Comment by paulvanbrenk** [comment]
@DanielRosenwasser, @bleuarg, or @Lagaffe does this look correct.

**Comment by RyanCavanaugh** [comment]
This isn't a correct fix. You need to **not** undo the UMD declaration (`export as namespace ...`).

File: types/nouislider/index.d.ts [hunk 1]
```diff
@@ -4,195 +4,196 @@
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 /// <reference types="wnumb"/>
 
-export = noUiSlider;
-export as namespace noUiSlider;
```
Comment (RyanCavanaugh) [inline types/nouislider/index.d.ts]: Don't remove this line

**Review by RyanCavanaugh — CHANGES_REQUESTED** [review]

File: types/nouislider/index.d.ts (hunk 1 above)
Comment (christophercr) [inline types/nouislider/index.d.ts]: You're right. I've added back the UMD declaration.

I also took the chance to add a missing option and use literal types where necessary ;)

**Comment by teppeis** [comment]
@RyanCavanaugh When this PR was merged, the CI job was failed and all packages updated after this merge are not published to npm.

- https://travis-ci.org/DefinitelyTyped/DefinitelyTyped/builds/253300526
- https://www.npmjs.com/package/@types/nouislider
- https://github.com/DefinitelyTyped/DefinitelyTyped/commits/master?after=c113d3ad62a3d4535b7493fccfc3c020c8d79487+69

**Comment by teppeis** [comment]
published

**Pull request merged**